webapp/src/main/java/com/box/l10n/mojito/service/branch/notification/BranchNotificationService.java

```diff
@@ -145,6 +147,16 @@ void sendNotificationsForBranchWithSender(
       BranchNotificationMessageSender branchNotificationMessageSender,
       Branch branch,
       BranchNotificationInfo branchNotificationInfo) {
+    // Check if the username for the Slack notification is in the block list
```

Comment:
Another way to do this without doing the `instanceof` check here would be to do the following:

In the interface for BranchNotificationMessageSender have `public boolean isNotificationAllowed/isUserAllowed`, while in the Slack implementation have an override that does the specific username/regex match (GH would always return true for that).

In this function then, we would only need to do this type of check:
```    
if (!branchNotificationMessageSender.isNotificationAllowed/isUserAllowed(branch)) {  
    return;
}  
```

What do you think?

Reply:
Done, thank you for the great suggestion!